test/sanitize_json.d

```diff
@@ -0,0 +1,231 @@
+        size_t inFileSize = cast(size_t)inFileSizeLong;
+        char[] buffer = new char[inFileSize + 1];
+        assert(inFileSize == inFile.rawRead(buffer[0..inFileSize]).length);
+        buffer[inFileSize] = '\0';
+        jsonText = cast(string)buffer;
+    }
+    auto json = parseJson(jsonText, inFilename);
+
+    sanitize(json.array);
+
+    auto writer = Writer(0);
+    writer.writeJson(json);
+    return 0;
+}
+
+void sanitize(JsonValue[] rootArray)
+{
+    foreach (objNode; rootArray)
+    {
+        auto obj = objNode.object;
+        auto kind = obj.getAs!string("kind");
+        if (kind == "compilerInfo")
+            sanitizeCompilerInfo(obj);
+        else if (kind == "buildInfo")
+            sanitizeBuildInfo(obj);
+        else if(kind == "module")
+            sanitizeSyntaxNode(objNode);
+        else if(kind == "semantics")
+            sanitizeSemantics(obj);
+    }
+}
+
+void removeString(JsonValue* value)
+{
+    assert(value.type == JsonType.string);
+    *value = JsonValue("VALUE_REMOVED_FOR_TEST");
+}
+void removeNumber(JsonValue* value)
+{
+    assert(value.type == JsonType.number);
+    *value = JsonValue(BigInt(0));
+}
+void removeStringIfExists(JsonObjectMemberReference member)
+{
+    if (!member.isNull)
+        removeString(member.value);
+}
+
+void sanitizeCompilerInfo(JsonObject buildInfo)
+{
+    removeString(buildInfo.getRef("binary").value);
+    removeString(buildInfo.getRef("version").value);
+}
+void sanitizeBuildInfo(JsonObject buildInfo)
+{
+    removeString(buildInfo.getRef("cwd").value);
+    removeStringIfExists(buildInfo.tryGetRef("config"));
+    removeStringIfExists(buildInfo.tryGetRef("lib"));
+}
+void sanitizeSyntaxNode(JsonValue value)
+{
+    if (value.type == JsonType.array)
+    {
+        foreach (ref element; value.arrayData)
+        {
+            sanitizeSyntaxNode(element);
+        }
+    }
+    else if(value.type == JsonType.object)
+    {
+        foreach (ref member; value.objectData.members)
+        {
+            if (member.name == "file")
+                removeString(&member.value);
+            else if (member.name == "deco")
+                removeString(&member.value);
+            else if (member.name == "offset")
+                removeNumber(&member.value);
+            else
+                sanitizeSyntaxNode(member.value);
+        }
+    }
+}
+
+void sanitizeSemantics(JsonObject semantics)
+{
+    auto modulesNode = semantics.getRef("modules");
+    auto modulesArray = modulesNode.value.array;
+    auto newModules = appender!(JsonValue[])();
+    foreach (ref semanticModuleNode; modulesArray)
+    {
+        auto semanticModule = semanticModuleNode.object;
+        auto moduleName = semanticModule.getAs!string("name");
+        if(moduleName.startsWith("std.") ||
+           moduleName.startsWith("core.") ||
+           moduleName.startsWith("etc.") ||
+           moduleName == "object")
+        {
+           continue;
+        }
```

Comment:
Without the special casing for Windows:

```d
modulesArray
.map!(m => object)
.filter!(m => !m.getAs!string("name").startsWith("std.", "core.", "etc."))
.each!(m => newModules.put(JsonValue(m)));
```

Reply:
My brain doesn't think in these algorithm functions very well :(  Is this easier for you to understand than the current loop?